core/net.go

```diff
@@ -387,11 +434,23 @@ func (n *OpenBazaarNode) SendOrderCompletion(peerID string, k *libp2p.PubKey, co
 	a, err := ptypes.MarshalAny(completionMessage)
 	if err != nil {
 		return err
-	}
-	return n.sendMessage(peerID, k, pb.Message{
+	}  
+	m := pb.Message{
 		MessageType: pb.Message_ORDER_COMPLETION,
 		Payload:     a,
-	})
+	}  
+	orderID0, err := n.CalcOrderID(completionMessage.BuyerOrder)
+	if err != nil {
+		log.Errorf("failed calculating order id")
```

Comment:
@amangale Doing some testing around this code, I hit this error and the message never saved: https://github.com/OpenBazaar/openbazaar-go/pull/1584/files#diff-8c006b1217c14444730603927b6f7515R444

Please add the error output to the log: `log.Errorf("failed calculating order id: %s", err)`.

Reply:
Looking at the call into this function, the `completionMessage.BuyerOrder` isn't populated which is causing this to fail. You may want to populate the BuyerOrder on the `completionMessage` in the calling function to fix this error. The `completionMessage` is created at `core/completion.go:244.

Reply:
In place of `CalcOrderID`, I have used `orderID0 := fulfillmentMessage.VendorOrderFulfillment[0].OrderId`.